bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java

```diff
@@ -195,161 +186,175 @@ public void dispose() {
+                            }
+                            if (value.equals("true")) {
+                                updateState(channel, OnOffType.ON);
+                            }
+                            channelValuesCounterConfigurable++;
+                            break;
+                        case 2:
+                            if (value.equals("false")) {
+                                State stateFalse = new StringType("0");
+                                updateState(channel, stateFalse);
+                            }
+                            if (value.equals("true")) {
+                                State stateTrue = new StringType("1");
+                                updateState(channel, stateTrue);
+                            }
+                            channelValuesCounterConfigurable++;
+                            break;
+                        case 3:
+                            State stateOther = getState(channelPostsConfigurable[channelValuesCounterConfigurable],
+                                    cConfig.unit);
+                            updateState(channel, stateOther);
+                            channelValuesCounterConfigurable++;
+                            break;
+                    }
                 }
             }
+        } catch (final RuntimeException e) {
+            logger.debug("Updating inverter status failed: ", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
         }
     }
 
     // Help method of handleCommand to with SecondGenerationConfigurationHandler.executeConfigurationChanges method send
     // configuration changes.
-    private final void preSetExecuteConfigurationChanges(HttpClient httpClient, String url, String username,
-            String password, String dxsEntriesConf, String valueConfiguration) {
+    private final void preSetExecuteConfigurationChanges(HttpClient httpClientHandleCommand, String url,
+            String username, String password, String dxsEntriesConf, String valueConfiguration) {
+
         try {
-            SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
-                    dxsEntriesConf, valueConfiguration);
-        } catch (Exception handleCommandException) {
-            logger.debug("Handle command for {} on channel {}: {}: {}: {}: {}", thing.getUID(), httpClient, url,
-                    dxsEntriesConf, valueConfiguration, handleCommandException.getMessage());
+            SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClientHandleCommand, url, username,
+                    password, dxsEntriesConf, valueConfiguration);
+        } catch (NoSuchAlgorithmException | InterruptedException | ExecutionException | TimeoutException e) {
+            logger.debug("Connection to inverter disturbed during configuration changes");
+            e.printStackTrace();
```

Comment:
Please do not do that. 
If you want to log the stack, just add your exception object as last parameter of your log.